src/gui/touchscreengui.cpp

```diff
@@ -106,7 +106,10 @@ static EKEY_CODE id_to_keycode(touch_gui_button_id id)
 		default:
 			break;
 	}
-	assert(!key.empty());
+	if (key.empty()) {
+		assert(!key.empty());
```

Comment:
IMHO the actions shouldn't be translated to keys. We'll have a problem when a user switches from the keyboard where he may deassign keys and then back to the touchscreen, where everything should work just fine.

Reply:
Yes, I agree, but that would require enough refactoring to consider it out of scope for this PR.

```cpp
	if (key.empty()) {
		assert(!key.empty());
```

What? ;)

Reply:
My approach regarding the unassigned key problem for this PR would be:

1. Only set a "rangeselect" key in `defaultsettings.cpp` if `ENABLE_TOUCH` is defined, i.e. if `enable_touch` is going to be enabled by default.

2. In `touchscreengui.cpp`, only show a button in the first place if the respective key is defined.